src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlCommand/SqlCommandCancelTest.cs

```diff
@@ -253,34 +257,99 @@ private static void TimeOutDuringRead(string constr)
-                    cmd.Parameters.AddWithValue("p", new byte[20000]);
-                    SqlDataReader reader = cmd.ExecuteReader();
-                    reader.Read();
-
-                    // Tweak the timeout to 1ms, stop the proxy from proxying and then try GetValue (which should timeout)
-                    reader.SetDefaultTimeout(1);
-                    proxy.PauseCopying();
-                    string errorMessage = SystemDataResourceManager.Instance.SQL_Timeout_Execution;
-                    Exception exception = Assert.Throws<SqlException>(() => reader.GetValue(0));
-                    Assert.Contains(errorMessage, exception.Message);
-
-                    // Return everything to normal and close
-                    proxy.ResumeCopying();
-                    reader.SetDefaultTimeout(30000);
-                    reader.Dispose();
+                    using (SqlCommand cmd = new SqlCommand("SELECT @p", conn))
+                    {
+                        cmd.Parameters.AddWithValue("p", new byte[20000]);
+                        using (SqlDataReader reader = cmd.ExecuteReader())
+                        {
+                            reader.Read();
+
+                            // Tweak the timeout to 1ms, stop the proxy from proxying and then try GetValue (which should timeout)
+                            reader.SetDefaultTimeout(1);
+                            proxy.PauseCopying();
+                            string errorMessage = SystemDataResourceManager.Instance.SQL_Timeout_Execution;
+                            Exception exception = Assert.Throws<SqlException>(() => reader.GetValue(0));
+                            Assert.Contains(errorMessage, exception.Message);
+
+                            // Return everything to normal and close
+                            proxy.ResumeCopying();
+                            reader.SetDefaultTimeout(30000);
+                            reader.Dispose();
+                        }
+                    }
                 }
-
-                proxy.Stop();
             }
             catch
             {
                 // In case of error, stop the proxy and dump its logs (hopefully this will help with debugging
                 proxy.Stop();
                 Console.WriteLine(proxy.GetServerEventLog());
-                Assert.True(false, "Error while reading through proxy");
                 throw;
             }
+            finally
+            {
+                proxy.Stop();
+            }
+        }
+
+        [CheckConnStrSetupFact]
+        public static void CancelDoesNotWait()
+        {
+            const int delaySeconds = 30;
+            const int cancelSeconds = 1;
+
+            using (SqlConnection conn = new SqlConnection(s_connStr))
+            using (var cmd = new SqlCommand($"WAITFOR DELAY '00:00:{delaySeconds:D2}'", conn))
+            {
+                conn.Open();
+
+                Task.Delay(TimeSpan.FromSeconds(cancelSeconds * 2))
```

Comment:
Since we are just trying to increase cancelSeconds so that the elapsed time in the last Assert is > cancelSeconds, I'd just add 1 instead of multiplying by 2. Also add a comment about why it is there since it's not obvious.